fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java

```diff
@@ -1156,6 +1157,16 @@ private Function checkPolymorphicFunction(Function fn, Type[] paramTypes) {
                     LOGGER.warn("could not determine polymorphic type because input has non-match types");
                     return null;
                 }
+            } else if (declType instanceof AnyMapType) {
+                if (realType.isNull()) {
+                    continue;
+                }
+                if (typeMap == null) {
+                    typeMap = (MapType) realType;
+                } else if ((typeMap = (MapType) getSuperType(typeMap, realType)) == null) {
```

Comment:
I think this step is used to check the function all parameters are the same type(including nested types in complex type), so our function must apply the totally same type for complex type?

Reply:
up to now, it's impossible, i will rewrite it.